keywrap/jwe/keywrapper_jwe.go

```diff
@@ -123,9 +123,13 @@ func addPubKeys(joseRecipients *[]jose.Recipient, pubKeys [][]byte) error {
 		}
 
 		alg := jose.RSA_OAEP
-		switch key.(type) {
+		switch key := key.(type) {
 		case *ecdsa.PublicKey:
 			alg = jose.ECDH_ES_A256KW
+		case *jose.JSONWebKey:
+			if key.Algorithm != "" {
+				alg = jose.KeyAlgorithm(key.Algorithm)
```

Comment:
This now opens up usage also for symmetric keys that the rest of the code cannot handle: https://github.com/go-jose/go-jose/blob/v3/shared.go#L90

I think we need another switch statement here that only accepts algorithms that we know we can handle which probably (subject to testing) are: ED25519, RSA*,  and ECDH_ES*

Reply:
ECDH_ES is not accepted in multi-recipient modes it seem, so this cannot be used, and it has to be ECDH_ES_A*
It's better to only use OAEP with RSA, so no RSA1_5.

Reply:
I think this here would resolve the issue:
```
diff --git a/keywrap/jwe/keywrapper_jwe.go b/keywrap/jwe/keywrapper_jwe.go
index 62e03b5..f281b08 100644
--- a/keywrap/jwe/keywrapper_jwe.go
+++ b/keywrap/jwe/keywrapper_jwe.go
@@ -129,6 +129,18 @@ func addPubKeys(joseRecipients *[]jose.Recipient, pubKeys [][]byte) error {
                case *jose.JSONWebKey:
                        if key.Algorithm != "" {
                                alg = jose.KeyAlgorithm(key.Algorithm)
+                               switch alg {
+                               /* accepted algorithms */
+                               case jose.ED25519:
+                               case jose.RSA_OAEP:
+                               case jose.RSA_OAEP_256:
+                               case jose.ECDH_ES_A128KW:
+                               case jose.ECDH_ES_A192KW:
+                               case jose.ECDH_ES_A256KW:
+                               /* all others are rejected */
+                               default:
+                                       return fmt.Errorf("%s is an unsupported JWK key algorithm", alg);
+                               }
                        }
                }

```

Reply:
> Actually, `parseJWKPublicKey` will never return a non-asymmetric key, because it checks for `IsPublic` on the JWK.
> 
> https://github.com/containers/ocicrypt/blob/a24b47713b8af833f6c389420b7e3b25f576554e/utils/utils.go#L53-L55
> 
> Fair point about multi-recipient mode, I had not considered that..

Yes, I have seen that but it will then go to try to parse it as a pkcs11 key and one gets odd error messages about a pkcs11 key that cannot be parse:

https://github.com/containers/ocicrypt/blob/a24b47713b8af833f6c389420b7e3b25f576554e/utils/utils.go#L150-L153

Reply:
I wasn't successful with ED25519 key  - it seems to get rejected by the MultiEcnrypter, so I had to take this out from the supported algorithms.